su1/concurrent-write.go

```diff
@@ -0,0 +1,68 @@
+package su1
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+)
+
+func CreateFile(fileName string) *os.File {
+	file, err := os.Create(fileName)
+
+	if err != nil {
+		fmt.Println("Error writing to file:", err)
+		return nil
+	}
+
+	fmt.Println("File created successfully")
+
+	return file
+}
+
+func WriteWithOffset(file *os.File, index int, chunk string, mu *sync.Mutex) {
```

Comment:
I believe we should provide the `offset` as an argument and pass the already calculated value since the function is called `WriteWithOffset`.